tachyon/zk/plonk/prover/witness_collection.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_PROVER_WITNESS_COLLECTION_H_
+#define TACHYON_ZK_PLONK_PROVER_WITNESS_COLLECTION_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "absl/container/btree_map.h"
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/base/range.h"
+#include "tachyon/zk/plonk/circuit/assignment.h"
+#include "tachyon/zk/plonk/circuit/phase.h"
+
+namespace tachyon::zk {
+
+template <typename PCSTy>
+class WitnessCollection : public Assignment<typename PCSTy::Field> {
+ public:
+  constexpr static size_t kMaxDegree = PCSTy::kMaxDegree;
+
+  using F = typename PCSTy::Field;
+  using Evals = typename PCSTy::Evals;
+  using RationalEvals = typename PCSTy::RationalEvals;
+  using AssignCallback = typename Assignment<F>::AssignCallback;
+
+  WitnessCollection() = default;
+  WitnessCollection(size_t k, size_t num_advice_columns,
+                    size_t unusable_row_start, const Phase current_phase,
```

Comment:
This looks to me clearer.

```suggestion
                    size_t usable_rows, const Phase current_phase,
```